rnahybrid.gap

```diff
@@ -705,6 +707,154 @@ algebra alg_probing implements sig_rnahybrid(alphabet = char, answer = double) {
+  int nil(<Subsequence qregion, Subsequence tregion>) {
+    return 0;
+  }
+  int target_left_flank(<Subsequence tregion, Subsequence mloc>, int x) {
+    return x;
+  }
+  int ulb(<Subsequence qloc, Subsequence tbase>, int x) {
+    return x;
+  }
+  int eds(<Subsequence qbase, Subsequence tbase>, int x) {
+    return x;
+  } 
+  int edt(<Subsequence qbase, Subsequence tloc>, int x) {
+    return x;
+  }
+  int edb(<Subsequence qloc, Subsequence tbase>, int x) {
+    return x;
+  }
+  int sr(<Subsequence qbase, Subsequence tbase>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res + 1;
+    }
+    return res;
+  }  
+  int bt(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tloc>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res * -1;
+    }
+    return res;
+  }
+  int bb(<Subsequence qbase, Subsequence tbase>, <Subsequence qloc, Subsequence tregion>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res * -1;
+    }
+    return res;
+  }
+  int il(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tregion>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res * -1;
+    }
+    return res;
+  }
+  int el(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tregion>) {
+    return 1;
+  }
+  int complete(int x) {
+    if (x < 0) {
+      return -1 * x;
+    } else {
+      return x;
+    }
+  }
+  choice [int] h([int] i) {
+    return unique(i);
+  }
+}
+algebra alg_khorshid implements sig_rnahybrid(alphabet = char, answer = khorshid) {
+  khorshid nil(<Subsequence qregion, Subsequence tregion>) {
+    khorshid res;
+    res.leftstacklen = 0;
+    res.mirnabuldgelen = 0;
+    return res;
+  }
+  khorshid target_left_flank(<Subsequence tregion, Subsequence mloc>, khorshid x) {
+    return x;
+  }
+  khorshid ulb(<Subsequence qloc, Subsequence tbase>, khorshid x) {
+    return x;
+  }
+  khorshid eds(<Subsequence qbase, Subsequence tbase>, khorshid x) {
+    return x;
+  } 
+  khorshid edt(<Subsequence qbase, Subsequence tloc>, khorshid x) {
+    return x;
+  }
+  khorshid edb(<Subsequence qloc, Subsequence tbase>, khorshid x) {
+    return x;
+  }
+  khorshid sr(<Subsequence qbase, Subsequence tbase>, khorshid x) {
+    khorshid res;
+    res.leftstacklen = x.leftstacklen;
+    res.mirnabuldgelen = x.mirnabuldgelen;
+    if (res.leftstacklen >= 0) {
+      res.leftstacklen = res.leftstacklen + 1;
+    }
+    return res;
+  }  
+  khorshid bt(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tloc>, khorshid x) {
+    khorshid res;
+    res.leftstacklen = x.leftstacklen;
+    res.mirnabuldgelen = x.mirnabuldgelen;
+    if (res.leftstacklen >= 0) {
+      res.leftstacklen = res.leftstacklen * -1;
+    }
+    return res;
+  }
+  khorshid bb(<Subsequence qbase, Subsequence tbase>, <Subsequence qloc, Subsequence tregion>, khorshid x) {
+    khorshid res;
+    res.leftstacklen = x.leftstacklen;
```

Comment:
Can this not be shortcut to "res = x" since we are transferring all relevant info anyways?

Reply:
good catch

Reply:
I fixed it at multiple places, thanks!